### PR TITLE
fix the driver pod can not be created due to unreasonable admit

### PR DESCRIPTION
**Pull request by william-wang**
Resolve the first item of #2040.  Also for #1778
Signed-off-by: william-wang <wang.platform@gmail.com>

### CODE REVIEW

**Comment by william-wang** [comment]
### Test case1: PodDelayCreation works as expected for vcjob, Volcano can not create pods whose podgroup is in pending state.
[root@ecs-ci yaml]# cat test-vcjob.yaml 
apiVersion: batch.volcano.sh/v1alpha1
kind: Job
metadata:
  name: test-vcjob
spec:
  plugins:
    svc: []
  schedulerName: volcano
  minAvailable: 5
  tasks:
    - replicas: 5
      name: "test"
      template:
        spec:
          containers:
            - image: alpine
              command: ["/bin/sh", "-c", "sleep 1000"]
              imagePullPolicy: IfNotPresent
              name: running
              resources:
                requests:
                  cpu: "1"
          restartPolicy: OnFailure
[root@ecs-ci yaml]# kubectl apply -f test-vcjob.yaml 

[root@ecs-ci yaml]# kubectl get pod
NAME         READY   STATUS    RESTARTS   AGE

Notes: The pod is not created due to it's podgroup is in pending state.

### Test case2: Pod allowed to be created for non-vcjob and Pods keep pending due to it's podgroup is in pending.
[root@ecs-ci yaml]# cat test-non-vcjob.yaml 
apiVersion: batch/v1
kind: Job
metadata:
  name: qj-1
spec:
  backoffLimit: 1
  completions: 3
  parallelism: 3
  template:
    metadata:
      annotations:
        scheduling.k8s.io/group-name: qj-1
    spec:
      containers:
      - image: busybox
        imagePullPolicy: IfNotPresent
        name: busybox
        resources:
          requests:
            cpu: "1"
      restartPolicy: Never
      schedulerName: volcano 

apiVersion: scheduling.volcano.sh/v1beta1
kind: PodGroup
metadata:
  name: qj-1
spec:
  minMember: 3
  minResources:
    cpu: 4500m
[root@ecs-ci yaml]# kubectl apply -f test-non-vcjob.yaml
[root@ecs-ci yaml]# kubectl get pod
NAME         READY   STATUS    RESTARTS   AGE
qj-1-dm659   0/1     Pending   0          8h
qj-1-dp9xz   0/1     Pending   0          8h
qj-1-jrhdh   0/1     Pending   0          8h

**Review by Yikun — APPROVED** [review]
I test spark master branch with volcano `release-1.5` branch + cherry-pick this patch .

```
[info] VolcanoSuite:
[info] - Run SparkPi with no resources (10 seconds, 986 milliseconds)
[info] - Run SparkPi with no resources & statefulset allocation (10 seconds, 651 milliseconds)
[info] - Run SparkPi with a very long application name. (10 seconds, 632 milliseconds)
[info] - Use SparkLauncher.NO_RESOURCE (10 seconds, 653 milliseconds)
[info] - Run SparkPi with a master URL without a scheme. (10 seconds, 711 milliseconds)
[info] - Run SparkPi with an argument. (10 seconds, 667 milliseconds)
[info] - Run SparkPi with custom labels, annotations, and environment variables. (10 seconds, 706 milliseconds)
[info] - All pods have the same service account by default (9 seconds, 642 milliseconds)
[info] - Run extraJVMOptions check on driver (5 seconds, 645 milliseconds)
[info] - Run SparkRemoteFileTest using a remote data file (10 seconds, 821 milliseconds)
[info] - Verify logging configuration is picked from the provided SPARK_CONF_DIR/log4j2.properties (17 seconds, 385 milliseconds)
[info] - Run SparkPi with env and mount secrets. (19 seconds, 534 milliseconds)
[info] - Run PySpark on simple pi.py example (11 seconds, 739 milliseconds)
[info] - Run PySpark to test a pyfiles example (12 seconds, 860 milliseconds)
[info] - Run PySpark with memory customization (10 seconds, 717 milliseconds)
[info] - Run in client mode. (7 seconds, 336 milliseconds)
[info] - Start pod creation from template (10 seconds, 778 milliseconds)
[info] - SPARK-38398: Schedule pod creation from template (10 seconds, 783 milliseconds)
[info] - Test basic decommissioning (44 seconds, 167 milliseconds)
[info] - Test basic decommissioning with shuffle cleanup (44 seconds, 281 milliseconds)
[info] - Test decommissioning with dynamic allocation & shuffle cleanups (2 minutes, 42 seconds)
[info] - Test decommissioning timeouts (45 seconds, 131 milliseconds)
[info] - SPARK-37576: Rolling decommissioning (1 minute, 7 seconds)
[info] - Run SparkPi with volcano scheduler (10 seconds, 931 milliseconds)
[info] - SPARK-38187: Run SparkPi Jobs with minCPU (30 seconds, 729 milliseconds)
[info] - SPARK-38187: Run SparkPi Jobs with minMemory (32 seconds, 628 milliseconds)
[info] - SPARK-38188: Run SparkPi jobs with 2 queues (only 1 enabled) (16 seconds, 286 milliseconds)
[info] - SPARK-38188: Run SparkPi jobs with 2 queues (all enabled) (28 seconds, 492 milliseconds)
[info] - SPARK-38423: Run driver job to validate priority order (17 seconds, 477 milliseconds)
[info] Run completed in 12 minutes, 10 seconds.
[info] Total number of tests run: 29
[info] Suites: completed 1, aborted 0
[info] Tests: succeeded 29, failed 0, canceled 0, ignored 0, pending 0
[info] All tests passed.
[success] Total time: 759 s (12:39), completed 2022-3-13 9:35:18
```

@william-wang Thanks for making it solid!

**Review by wpeng102 — APPROVED** [review]
/lgtm

File: pkg/webhooks/admission/pods/validate/admit_pod.go [hunk 1]
```diff
@@ -96,9 +96,8 @@ func AdmitPods(ar admissionv1.AdmissionReview) *admissionv1.AdmissionResponse {
 /*
 allow pods to create when
 1. schedulerName of pod isn't volcano
-2. pod has Podgroup whose phase isn't Pending
```
Comment (shinytang6) [inline pkg/webhooks/admission/pods/validate/admit_pod.go:99]: /lgtm

**Review by shinytang6 — COMMENTED** [review]
/approve

File: pkg/webhooks/admission/pods/validate/admit_pod.go (hunk 1 above)
Comment (Yikun) [reply pkg/webhooks/admission/pods/validate/admit_pod.go:99]: Thanks for post review!

**Comment by BrushXiaoMinGuo** [comment]
@william-wang  I have a confusion about this modification. How this modification balances the delayed creation of pods to reduce the API server's load .

**Pull request merged**